### PR TITLE
[RF] Improve plotting or RooBinSamplingPdf

### PR DESCRIPTION
**Pull request by hageboeck**
Since the bin sampling PDF is a binned distribution, one can greatly improve plotting by implementing plot sampling hints.

### CODE REVIEW

File: roofit/roofitcore/src/RooBinSamplingPdf.cxx [hunk 1]
```diff
@@ -199,6 +199,56 @@ RooSpan<const double> RooBinSamplingPdf::binBoundaries() const {
...
+    return nullptr;
+  }
+
+  auto list = new std::list<double>;
+  for (double val : binBoundaries()) {
+    if (xlo <= val && val < xhi)
+      list->push_back(val);
+  }
+
+  return list;
```
Comment (jalopezg-uc3m) [inline roofit/roofitcore/src/RooBinSamplingPdf.cxx:221]: And who `delete`s this thing?

File: roofit/roofitcore/src/RooBinSamplingPdf.cxx [hunk 2]
```diff
@@ -199,6 +199,56 @@ RooSpan<const double> RooBinSamplingPdf::binBoundaries() const {
...
+  const auto& binning = obs.getBinning();
+
+  for (unsigned int bin=0; bin < static_cast<unsigned int>(binning.numBins()); ++bin) {
+    const double centre = binning.binCenter(bin);
+
+    if (xlo <= centre && centre < xhi)
+      binCentres->push_back(centre);
+  }
+
+  return binCentres;
```
Comment (jalopezg-uc3m) [inline roofit/roofitcore/src/RooBinSamplingPdf.cxx:248]: Same here.

File: roofit/roofitcore/src/RooBinSamplingPdf.cxx (hunk 1 above)
Comment (jalopezg-uc3m) [reply roofit/roofitcore/src/RooBinSamplingPdf.cxx:221]: Don't know how the interface is used in other places, but maybe it's better to use smart ptrs?

**Review by jalopezg-uc3m — COMMENTED** [review]
Check the pointer issue, but for the rest, it is approved.  :-)

**Review by jalopezg-uc3m — APPROVED** [review]

File: roofit/roofitcore/src/RooBinSamplingPdf.cxx (hunk 1 above)
Comment (hageboeck) [reply roofit/roofitcore/src/RooBinSamplingPdf.cxx:221]: I wish ...

As you can see, it's `override`, so I cannot change it ... The caller has to delete it.

**Pull request merged**